verifier/verifier.py

```diff
@@ -1,276 +1,335 @@
+                                                      self.max_samples,
+                                                      features,
+                                                      self.debug,
+                                                      self.use_gpu)
+
+                # Record time for class initialization
+                initialize_time = timeit.default_timer() - start
+
+                # Register times for asset processing
+                start = timeit.default_timer() - start
+
+                # Assemble output dataframe with processed metrics
+                metrics_df, pixels_df, dimensions_df = asset_processor.process()
+
+                # Record time for processing of assets metrics
+                process_time = timeit.default_timer() - start
+
+                x_renditions_sl = np.asarray(metrics_df[self.features_sl])
+                x_renditions_ul = np.asarray(metrics_df[self.features_ul])
+                x_renditions_ul = self.loaded_scaler.transform(x_renditions_ul)
+                x_renditions_qoe = np.asarray(metrics_df[self.features_qoe])
+
+                np.set_printoptions(precision=6, suppress=True)
+                logger.debug('INPUT SL ARRAY:', x_renditions_sl, flush=True)
+                logger.debug('Unscaled INPUT UL ARRAY:', np.asarray(metrics_df[self.features_ul]), flush=True)
+                logger.debug('SCALED INPUT UL ARRAY:', x_renditions_ul, flush=True)
+                logger.debug('INPUT QOE ARRAY:', x_renditions_qoe, flush=True)
+                # Make predictions for given data
+                start = timeit.default_timer()
+                predictions_df = pd.DataFrame()
+                predictions_df['sl_pred_tamper'] = self.loaded_model_sl.predict(x_renditions_sl)
+                predictions_df['ssim_pred'] = self.loaded_model_qoe.predict(x_renditions_qoe)
+                predictions_df['ocsvm_dist'] = self.loaded_model_ul.decision_function(x_renditions_ul)
+                predictions_df['ul_pred_tamper'] = self.loaded_model_ul.predict(x_renditions_ul)
+                predictions_df['meta_pred_tamper'] = predictions_df.apply(self.meta_model, axis=1)
+                prediction_time = timeit.default_timer() - start
+
+                # Add predictions to rendition dictionary
+                i = 0
+                for _, rendition in enumerate(renditions):
+                    if rendition['video_available']:
+                        rendition.pop('path', None)
+                        rendition['ssim_pred'] = float(predictions_df['ssim_pred'].iloc[i])
+                        rendition['ocsvm_dist'] = float(predictions_df['ocsvm_dist'].iloc[i])
+                        rendition['tamper_ul'] = int(predictions_df['ul_pred_tamper'].iloc[i])
+                        rendition['tamper_sl'] = int(predictions_df['sl_pred_tamper'].iloc[i])
+                        rendition['tamper'] = int(predictions_df['meta_pred_tamper'].iloc[i])
```

Comment:
Perhaps we can name this key "tamper_meta" to match the name used in the API response

Reply:
I think API (and even Verifier class) shouldn't expose much model implementation details (aside of debug purposes) to avoid confusion.